### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by anpiet**
advancedCalculator Solution

### CODE REVIEW

File: src/makeCalculator.js [hunk 1]
```diff
@@ -37,7 +37,46 @@
...
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(num) {
+      this.result += num;
+
+      return this;
+    },
```
Comment (AndrzejPec) [inline src/makeCalculator.js:47]: I guess you could shorten it with arrow funciton ▶️

File: src/makeCalculator.js [hunk 2]
```diff
@@ -37,7 +37,46 @@
...
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, num) {
+      callback.call(this, num);
+
+      return this;
```
Comment (AndrzejPec) [inline src/makeCalculator.js:76]: same as above - but I'm not sure.

**Review by AndrzejPec — CHANGES_REQUESTED** [review]
little improvement suggested, good job 🎸 🥁

File: src/makeCalculator.js (hunk 1 above)
Comment (anpiet) [reply src/makeCalculator.js:47]: It's not possible to use arrow function here.

File: src/makeCalculator.js (hunk 2 above)
Comment (anpiet) [reply src/makeCalculator.js:76]: It's not possible to use arrow function here.

File: src/makeCalculator.js (hunk 1 above)
Comment (JoyCoffeeAddict) [reply src/makeCalculator.js:47]: @AndrzejPec That is not possible, because `this` keyword has different value within arrow functions.

**Review by JoyCoffeeAddict — APPROVED** [review]

**Review by choeqq — APPROVED** [review]

**Pull request closed without merging**